lib/puppet/util/execution.rb

```diff
@@ -187,18 +187,30 @@ def self.execute(command, options = NoOptionsSpecified)
     null_file = Puppet.features.microsoft_windows? ? 'NUL' : '/dev/null'
 
     begin
+      reader, writer = IO.pipe unless options[:squelch]
```

Comment:
Full explanation of closing a process in https://msdn.microsoft.com/en-us/library/windows/desktop/ms686714(v=vs.85).aspx.